frame/support/src/traits/hooks.rs

```diff
@@ -80,13 +85,20 @@ pub trait OnIdle<BlockNumber> {
 }
 
 #[impl_for_tuples(30)]
-impl<BlockNumber: Clone> OnIdle<BlockNumber> for Tuple {
+impl<BlockNumber: 'static + Copy + AtLeast32BitUnsigned> OnIdle<BlockNumber> for Tuple {
+	for_tuples!( const ON_IDLE_FUNCTIONS : &'static [fn(BlockNumber, crate::weights::Weight) -> crate::weights::Weight] = &[ #(Tuple::on_idle),* ]; );
+
 	fn on_idle(n: BlockNumber, remaining_weight: crate::weights::Weight) -> crate::weights::Weight {
 		let mut weight = 0;
-		for_tuples!( #(
+		let len = <Self as OnIdle<BlockNumber>>::ON_IDLE_FUNCTIONS.len();
```

Comment:
```suggestion
		for_tuples!( const ON_IDLE_FUNCTIONS : &'static [fn(BlockNumber, crate::weights::Weight) -> crate::weights::Weight] = &[ #(Tuple::on_idle),* ]; );
		let len = <Self as OnIdle<BlockNumber>>::ON_IDLE_FUNCTIONS.len();
```

I think Basti means to add the const here in the function

Reply:
Yes we have discussed it, it such case it is not possible due to this error https://doc.rust-lang.org/error-index.html#E0401 and here it is a similar example with the same problem https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3db75848b9657a0649e02f9a641ad48c